controlplane/eks/controllers/awsmanagedcontrolplane_controller.go

```diff
@@ -57,11 +57,21 @@ const (
 	deleteRequeueAfter = 20 * time.Second
 )
 
-var (
-	eksSecurityGroupRoles = []infrav1.SecurityGroupRole{
-		infrav1.SecurityGroupEKSNodeAdditional,
+var defaultEKSSecurityGroupRoles = []infrav1.SecurityGroupRole{
+	infrav1.SecurityGroupEKSNodeAdditional,
+}
+
+// securityGroupRolesForControlPlane returns the security group roles determined by the control plane configuration.
+func securityGroupRolesForControlPlane(scope *scope.ManagedControlPlaneScope) []infrav1.SecurityGroupRole {
+	// Copy to ensure we do not modify the package-level variable.
+	roles := make([]infrav1.SecurityGroupRole, len(defaultEKSSecurityGroupRoles))
+	copy(roles, defaultEKSSecurityGroupRoles)
+
+	if scope.Bastion().Enabled {
+		roles = append(roles, infrav1.SecurityGroupBastion)
```

Comment:
```suggestion
	// Copy to ensure we do not modify the package-level variable.
	roles := append([]infrav1.SecurityGroupRole{}, infrav1.SecurityGroupEKSNodeAdditional)
	if scope.Bastion().Enabled {
		roles = append(roles, infrav1.SecurityGroupBastion)
```

Does the same thing, without the copy need. :)

Reply:
I see your point, but I think we want to keep `defaultEKSSecurityGroupRoles` as a reminder of the default set of security group roles.

Reply:
Sure, if you really want that, you can do:

```go
roles := append([]infrav1.SecurityGroupRole{}, defaultEKSSecurityGroupRoles...)
```

Which will also keep the original. :)

Reply:
@pydctw and I also considered using `append` here, but we settled on using `copy`. I'll resolve this thread for now, but if you feel very strongly, please reopen.

Reply:
It's fine. :) I'm just pedantic. :D

Reply:
No problem at all :smile:  I am also pedantic! ... Just not on my own PRs :joy: